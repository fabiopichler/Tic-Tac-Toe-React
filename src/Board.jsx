import React from 'react';

import { Row, Col } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSun, faMoon } from '@fortawesome/free-solid-svg-icons';

const Board = ({
    game,
    click,
}) => (
    <main className="board">
        <section className="board-container">
            {game.board.map((row, rowIndex) => (
                <Row
                    className="board-row"
                    key={rowIndex}
                >
                    {row.map((col, colIndex) => (
                        <Col
                            onClick={() => click(col, rowIndex, colIndex)}
                            className="board-col"
                            key={`${rowIndex}${colIndex}`}
                        >
                            {col === 1 ? (
                                <FontAwesomeIcon
                                    icon={faSun}
                                    className="board-player-icon player-1 text-warning"
                                />
                            ) : col === 2 ? (
                                <FontAwesomeIcon
                                    icon={faMoon}
                                    className="board-player-icon player-2 text-info"
                                />
                            ) : null}
                        </Col>
                    ))}
                </Row>
            ))}
        </section>
    </main>
);

export default Board;
