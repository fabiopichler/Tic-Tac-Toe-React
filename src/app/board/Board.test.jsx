import React from 'react';
import renderer from 'react-test-renderer';

import Board from './Board';

test('Board: renders correctly', () => {
    const component = renderer.create(
        <Board
            game={{
                player: 1,
                round: 0,
                winner: 0,
                board: [
                    [0, 0, 0],
                    [0, 0, 0],
                    [0, 0, 0],
                ],
            }}
            click={() => { }}
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
