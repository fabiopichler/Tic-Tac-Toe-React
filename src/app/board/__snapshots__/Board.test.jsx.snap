// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board: renders correctly 1`] = `
<main
  className="board"
>
  <section
    className="board-container"
  >
    <div
      className="board-row row"
    >
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
    </div>
    <div
      className="board-row row"
    >
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
    </div>
    <div
      className="board-row row"
    >
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
      <div
        className="board-col col"
        onClick={[Function]}
      />
    </div>
  </section>
</main>
`;
